### PR TITLE
Replace flutter.io -> flutter.dev in dart and java files.

### PR DESCRIPTION
**Pull request by GaryQian**
Since we are now switching to using flutter.dev, we should refer to the new site in the docs/comments/code/etc.

This is a partial conversion that changes all framework dart and java files.

I have skimmed this to make sure the places where we change the URL are reasonable.

The complete transition will involve slightly more technical examination to make sure we are not converting binary files or places where a conversion is invalid/impractical.

### CODE REVIEW

File: dev/docs/platform_integration/lib/android.dart [hunk 1]
```diff
@@ -1,2 +1,2 @@
-/// [Flutter platform integration APIs for Android.](https://docs.flutter.io/javadoc/)
+/// [Flutter platform integration APIs for Android.](https://docs.flutter.dev/javadoc/)
```
Comment (gspencergoog) [inline dev/docs/platform_integration/lib/android.dart]:
```suggestion
/// [Flutter platform integration APIs for Android.](https://api.flutter.dev/javadoc/)
```

File: dev/docs/platform_integration/lib/ios.dart [hunk 2]
```diff
@@ -1,2 +1,2 @@
-/// [Flutter platform integration APIs for iOS.](https://docs.flutter.io/objcdoc/)
+/// [Flutter platform integration APIs for iOS.](https://docs.flutter.dev/objcdoc/)
```
Comment (gspencergoog) [inline dev/docs/platform_integration/lib/ios.dart]:
```suggestion
/// [Flutter platform integration APIs for iOS.](https://api.flutter.dev/objcdoc/)
```

File: dev/tools/dartdoc.dart [hunk 3]
```diff
@@ -20,8 +20,8 @@ const String kSnippetsRoot = 'dev/snippets';
 /// documentation to `//dev/docs/doc/api/`.
 ///
 /// This script also updates the index.html file so that it can be placed
-/// at the root of docs.flutter.io. We are keeping the files inside of
-/// docs.flutter.io/flutter for now, so we need to manipulate paths
+/// at the root of docs.flutter.dev. We are keeping the files inside of
+/// docs.flutter.dev/flutter for now, so we need to manipulate paths
```
Comment (gspencergoog) [inline dev/tools/dartdoc.dart]:
```suggestion
/// api.flutter.dev/flutter for now, so we need to manipulate paths
```

File: dev/tools/dartdoc.dart [hunk 4]
```diff
@@ -20,8 +20,8 @@ const String kSnippetsRoot = 'dev/snippets';
 /// documentation to `//dev/docs/doc/api/`.
 ///
 /// This script also updates the index.html file so that it can be placed
-/// at the root of docs.flutter.io. We are keeping the files inside of
-/// docs.flutter.io/flutter for now, so we need to manipulate paths
+/// at the root of docs.flutter.dev. We are keeping the files inside of
```
Comment (gspencergoog) [inline dev/tools/dartdoc.dart]:
```suggestion
/// at the root of api.flutter.dev. We are keeping the files inside of
```

File: dev/tools/dartdoc.dart [hunk 5]
```diff
@@ -284,7 +284,7 @@ void createSearchMetadata(String templatePath, String metadataPath) {
   final String branch = getBranchName();
   final String metadata = template.replaceAll(
     '{SITE_URL}',
-    branch == 'stable' ? 'https://docs.flutter.io/' : 'https://master-docs.flutter.io/',
+    branch == 'stable' ? 'https://docs.flutter.dev/' : 'https://master-docs.flutter.dev/',
```
Comment (gspencergoog) [inline dev/tools/dartdoc.dart]:
```suggestion
    branch == 'stable' ? 'https://api.flutter.dev/' : 'https://master-api.flutter.dev/',
```

File: dev/docs/platform_integration/lib/android.dart (hunk 1 above)
Comment (gspencergoog) [reply dev/docs/platform_integration/lib/android.dart]: And below, everywhere...

File: dev/tools/dartdoc.dart (hunk 5 above)
Comment (gspencergoog) [reply dev/tools/dartdoc.dart]: And below, everywhere, for https://master-api.flutter.dev.

File: packages/flutter_tools/test/ios/mac_test.dart [hunk 6]
```diff
@@ -288,7 +288,7 @@ void main() {
         success: false,
         stdout: '''
 Launching lib/main.dart on iPhone in debug mode...
-Signing iOS app for device deployment using developer identity: "iPhone Developer: test@flutter.io (1122334455)"
+Signing iOS app for device deployment using developer identity: "iPhone Developer: test@flutter.dev (1122334455)"
```
Comment (gspencergoog) [inline packages/flutter_tools/test/ios/mac_test.dart]: @kf6gpe, are the MX records set up to do mail correctly to .dev too?  This is just text, so it doesn't really do anything, but the thought just occured to me...

**Review by gspencergoog — CHANGES_REQUESTED** [review]

**Comment by GaryQian** [comment]
FYI, api.flutter.dev is currently a redirect to docs.flutter.io

**Comment by gspencergoog** [comment]
Yeah, but that will change. @kf6gpe is doing that.

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

File: dev/docs/platform_integration/lib/android.dart [hunk 7]
```diff
@@ -1,2 +1,2 @@
-/// [Flutter platform integration APIs for Android.](https://docs.flutter.io/javadoc/)
+/// [Flutter platform integration APIs for Android.](https://api.flutter.dev/javadoc/)
```
Comment (DanTup) [inline dev/docs/platform_integration/lib/android.dart]: I don't believe the redirects handle this properly - these two URLs are not the same - the .dev one is redirecting to the root (and non-folder URLs on the new domain are 404!).

File: examples/flutter_gallery/lib/demo/material/buttons_demo.dart [hunk 8]
```diff
@@ -70,7 +70,7 @@ class _ButtonsDemoState extends State<ButtonsDemo> {
           child: buildRaisedButton(),
         ),
         exampleCodeTag: _raisedCode,
-        documentationUrl: 'https://docs.flutter.io/flutter/material/RaisedButton-class.html',
+        documentationUrl: 'https://api.flutter.dev/flutter/material/RaisedButton-class.html',
```
Comment (DanTup) [inline examples/flutter_gallery/lib/demo/material/buttons_demo.dart]: eg. this one is currently a 404. Though if you're not planning on landing this until the domains are swapped over properly, that's probably not an issue.

**Comment by gspencergoog** [comment]
Sorry, I thought we had a DNS alias already in place for this, not just the HTML redirect.  I agree, don't commit this yet.

**Comment by kf6gpe** [comment]
Yes. I'm thinking we can actually point the new URL at the old firebase
project and then commit this, and when we publish to the new Firebase
project as part of our next stable release, just switch the back end
project.

Does that make sense?

**Comment by GaryQian** [comment]
Closing this, as lots of conflicts means it is probably easier to just remake this PR.

**Pull request closed without merging**